internal/interactivity/interactiveModel.go

```diff
@@ -0,0 +1,69 @@
+package interactivity
+
+import (
+	"fmt"
+	tea "github.com/charmbracelet/bubbletea"
+	"os"
+)
+
+type InteractiveModel interface {
+	Init()
+	HandleKeypress(key string) bool // return true to signal exit
+	Render() string
+	SetWidth(width int)
+}
+
+type interactiveModelInternal struct {
+	model      InteractiveModel
+	signalExit bool
+}
+
+func (m interactiveModelInternal) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch keypress := msg.String(); keypress {
+		case "ctrl+c", "esc":
+			m.signalExit = true
+			return m, tea.Quit
+		default:
+			if m.model.HandleKeypress(keypress) {
+				return m, tea.Quit
+			}
+		}
+	case tea.WindowSizeMsg:
+		m.model.SetWidth(msg.Width)
+	}
+
+	return m, nil
+}
+
+func (m interactiveModelInternal) View() string {
+	return m.model.Render()
+}
+
+func (m interactiveModelInternal) Init() tea.Cmd {
+	m.model.Init()
+	return nil
+}
+
+func RunInteractiveModel(m InteractiveModel) InteractiveModel {
```

Comment:
Can we rename this. In think it's going to get confused with interactive mode of the CLI. This would be used by any tea model we have.

Maybe just RunModel and InternalModel. Not willing to fight for it though.